### ISSUE TITLE
NewCondition构造器使用AndCond方法引用自身，越过panic检测

### ISSUE THREAD

**Issue by gswcodeshow** [astaxie/beego#4298]
Please answer these questions before submitting your issue. Thanks!

1. What version of Go and beego are you using (`bee version`)?
 beego@v1.12.2

2. What operating system and processor architecture are you using (`go env`)?
max/amd64

3. What did you do?
使用下面的代码，不会报panic，可以正常执行。
```
package main

import (
	"fmt"

	"github.com/astaxie/beego/orm"
)

func main() {
	cond := orm.NewCondition()
	cond = cond.And("id", 1)
	cond = cond.AndCond(cond)
	fmt.Println(cond)
	return
}

```

4. What did you expect to see?
正常打印了cond的内容。

5. What did you see instead?
panic(fmt.Errorf("<Condition.AndCond> cannot use self as sub cond"))

**Comment by gswcodeshow** [comment]
查看了修改，发现了clone方法，还会存在共享内存的问题。
测试代码：
````
package main

import (
	"fmt"
)

type condValue struct {
	cond   *Condition
	exprs  string
	isCond bool
}

type Condition struct {
	params []condValue
}

func (c Condition) clone() *Condition {
	// params := make([]condValue, len(c.params))
	// copy(params, c.params)
	// c.params = params
	return &c
}

func (c Condition) And(expr string) *Condition {
	c.params = append(c.params, condValue{exprs: expr})
	return &c
}

func (c *Condition) AndCond(cond *Condition) *Condition {
	c = c.clone()
	if c == cond {
		panic(fmt.Errorf("<Condition.AndCond> cannot use self as sub cond"))
	}
	if cond != nil {
		c.params = append(c.params, condValue{cond: cond, isCond: true})
	}
	return c
}

func getCondSQL(cond *Condition) (where string) {
	if cond == nil {
		return
	}
	for _, p := range cond.params {
		if p.isCond {
			w := getCondSQL(p.cond)
			if w != "" {
				w = fmt.Sprintf("( %s ) ", w)
			}
			where += w
		} else {
			exprs := p.exprs
			where += exprs
		}
	}
	return
}

func main() {
	cond := &Condition{}
	cond = cond.AndCond(cond.And("1"))
	cond = cond.AndCond(cond.And("2"))
	cond = cond.AndCond(cond.And("3"))
	cond = cond.AndCond(cond.And("4"))
	condStr := getCondSQL(cond)
	fmt.Println(condStr)
	return
}

````
对clone进行了，值拷贝的修改。
能否解决这类的问题。

**Comment by gswcodeshow** [comment]
Fix 4298 #4304
